src/jquery.payment.coffee

```diff
@@ -104,7 +111,10 @@ $.payment.cards = cards = [
 
 cardFromNumber = (num) ->
   num = (num + '').replace(/\D/g, '')
-  return card for card in cards when card.pattern.test(num)
+  for card in cards
+    for pattern in card.patterns
+      r = new RegExp("^#{pattern}")
+      return card if r.exec(num)?
```

Comment:
This is too heavy a hammer. If you want to test that the string begins with a pattern, you should use `str.indexOf(pattern) === 0` instead.